src/System.Private.CoreLib/shared/System/Security/SecureString.Unix.cs

```diff
@@ -20,307 +20,19 @@ namespace System.Security
 
     public sealed partial class SecureString
     {
-        private UnmanagedBuffer? _buffer;
-
-        internal SecureString(SecureString str)
-        {
-            // Allocate enough space to store the provided string
-            EnsureCapacity(str._decryptedLength);
-            _decryptedLength = str._decryptedLength;
-
-            // Copy the string into the newly allocated space
-            if (_decryptedLength > 0)
-            {
-                Debug.Assert(str._buffer != null && _buffer != null); ;
-                UnmanagedBuffer.Copy(str._buffer, _buffer, (ulong)(str._decryptedLength * sizeof(char)));
-            }
-        }
-
-        private unsafe void InitializeSecureString(char* value, int length)
-        {
-            // Allocate enough space to store the provided string
-            EnsureCapacity(length);
-            _decryptedLength = length;
-            if (length == 0)
-            {
-                return;
-            }
-
-            Debug.Assert(_buffer != null);
-            // Copy the string into the newly allocated space
-            byte* ptr = null;
-            try
-            {
-                _buffer.AcquirePointer(ref ptr);
-                Buffer.MemoryCopy(value, ptr, _buffer.ByteLength, (ulong)(length * sizeof(char)));
-            }
-            finally
-            {
-                if (ptr != null)
-                {
-                    _buffer.ReleasePointer();
-                }
-            }
-        }
-
-        private void DisposeCore()
-        {
-            if (_buffer != null && !_buffer.IsInvalid)
-            {
-                _buffer.Dispose();
-                _buffer = null;
-            }
-        }
-
-        private void ClearCore()
+        private static int GetAlignedByteSize(int length)
         {
-            _decryptedLength = 0;
-            Debug.Assert(_buffer != null);
-            _buffer.Clear();
+            return Math.Max(length, 1) * sizeof(char);
         }
 
-        private unsafe void AppendCharCore(char c)
+        private void ProtectMemory()
         {
-            // Make sure we have enough space for the new character, then write it at the end.
-            EnsureCapacity(_decryptedLength + 1);
-            Debug.Assert(_buffer != null);
-            _buffer.Write((ulong)(_decryptedLength * sizeof(char)), c);
-            _decryptedLength++;
+            _encrypted = true;
         }
 
-        private unsafe void InsertAtCore(int index, char c)
+        private void UnprotectMemory()
         {
-            // Make sure we have enough space for the new character, then shift all of the characters above it and insert it.
-            EnsureCapacity(_decryptedLength + 1);
-            byte* ptr = null;
-            Debug.Assert(_buffer != null);
-            try
-            {
-                _buffer.AcquirePointer(ref ptr);
```

Comment:
The Windows and Unix versions had a lot of style differences. The only real difference were the extra UnprotectMemory/ProtectMemory calls in the Windows versions. This is handled by empty UnprotectMemory/ProtectMemory on Unix now.